src/System.Private.CoreLib/shared/System/Reflection/Assembly.cs

```diff
@@ -148,6 +148,11 @@ public override string ToString()
         public override bool Equals(object o) => base.Equals(o);
         public override int GetHashCode() => base.GetHashCode();
 
+        // Non-inline call to the virtual Equals so operator== only inlines to the ReferenceEquals 
```

Comment:
I do not think this is worth it. I am not concerned about a small increase of code size from this change.

Reply:
This no-inlining can actually hurt - by inhibiting potential de-virtualization.

Reply:
Removed